wordpress-mobile/AztecEditor-Android
# Issue/618 onMediaDeleted called unnecessarily in sequence

### Fix #618 

This PR implements a small (500ms) wait on the call to `onMediaDeleted` callback to make sure that, if an `AztecMediaSpan` has been deleted, it has been due to a user-triggered action rather than an event generated by the platform as discovered in API26+.

The check makes sense by identifying whether the `ObservationQueue` is being populated (with events that are very close together in time). If it's not the case, we can have a good amount of confidence that this is a user-triggered event, and so we end up triggering the actual `onMediaDeleted` events collected in the meantime.

Note that the 100ms time distance in between events is the one as already being used in `API26InWordSpaceInsertionEvent`. This value is being taken over to `ObservationQueue` on another PR. As both PRs are merged, we can get rid of this value declaration [here](https://github.com/wordpress-mobile/AztecEditor-Android/compare/issue/618-onimagedeleted-called-in-sequence?expand=1#diff-7ca71cd1f009ab8438e0901205001b14R857) and make use of the constant defined in `ObservationQueue` in `https://github.com/wordpress-mobile/AztecEditor-Android/pull/619/files#diff-5c63c1be1a17a9099fee1cf7d4476ea4R82`


### Test
On API26/27, using the demo app:

TEST A:
1. anywhere in the text, insert a line and type some letters, for example "hel".
2. now toggle Bold
3. now type "lo" (so the "lo" part is styled in bold)
4. place the cursor right before "lo", it should be seen like this at this position: "hel|**lo**""
5. now press SPACE - this will trigger the event sequence the `ObservationQueue` is watching for on API26+
6. Observe no "Media Deleted!" Toasts are shown.

TEST B:
1. now place the cursor right after any of the images in the demo app content
2. press BACKSPACE
3. observe the "Media Deleted!" Toast is shown accordingly.

TEST C:
On any other supported API, using the demo app, repeat TEST A.


### Review
@daniloercoli @0nko
